cli/internal/publish/images.go

```diff
@@ -0,0 +1,367 @@
+package publish
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha1"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
+
+	cloudquery_api "github.com/cloudquery/cloudquery-api-go"
+	"github.com/cloudquery/cloudquery/cli/internal/hub"
+	"github.com/yuin/goldmark"
+	"github.com/yuin/goldmark/ast"
+	"github.com/yuin/goldmark/parser"
+	"github.com/yuin/goldmark/text"
+	"github.com/yuin/goldmark/util"
+	"golang.org/x/sync/errgroup"
+)
+
+type imageReference struct {
+	ref     string // image filename (to replace with URL)
+	absFile string // absolute path to image file, to upload
+	url     string // result of upload
+
+	startPos, endPos int
+}
+
+var htmlImageRe = regexp.MustCompile(`<img\s+(?:[^>]*?\s+)?src="([^"]*)"`)
```

Comment:
Am I right in thinking there are some cases this can't handle yet, like `<img title="<important>" src="sda">` and maybe `<div><img src="image1.png"><img src="image2.png"></div>`?

Reply:
We could use https://pkg.go.dev/golang.org/x/net/html maybe

Reply:
I don't think we should make the effort to support that first one, it's supposed to be a markdown document after all.

<img title="<important>" src="https://storage.googleapis.com/cq-cloud-images/kemal/6027b3c3f2afcb2cf2c9405637a40690adb9a675_troll.png">

Does github support it?
And the second one might just work, but it's still a lot of html in a markdown. Using `<img>` in markdown is allowed only because there are some attributes you might want to set, like `width`.

Reply:
First case fixed in https://github.com/cloudquery/cloudquery/pull/16033/commits/56421e7a110bc2e6587e132aa8efe35d0230d970
Second case was previously fixed

Reply:
@hermanschaaf Using x/net/html as of https://github.com/cloudquery/cloudquery/pull/16033/commits/015d93250a0e8aa498f7e712e32592b6e997725b